lighthouse-core/gather/drivers/extension.js

```diff
@@ -152,6 +152,22 @@ class ExtensionDriver extends Driver {
   getCurrentTabURL() {
     return this.queryCurrentTab_().then(tab => tab.url);
   }
+
+  getCurrentTabId() {
+    return new Promise((resolve, reject) => {
+      this.queryCurrentTab_().then(currentTab => {
+        chrome.debugger.getTargets(targets => {
+          const target = targets.find(target => target.tabId === currentTab.id);
+
+          if (!target) {
+            reject(new Error('We can\'t find a target id.'));
+          }
+
+          resolve(target.id);
+        });
+      });
+    });
+  }
```

Comment:
I think you want to switch the outer Promise and `this.queryCurrentTab_` here, to put `this.queryCurrentTab_()` into the promise chain so any rejections in it will be handled. Something like:
```js
return this.queryCurrentTab_().then(currentTab => {
  return new Promise((resolve, reject) => {
    chrome.debugger.getTargets(targets => {
      const target = targets.find(target => target.tabId === currentTab.id);

      if (!target) {
        reject(new Error('We can\'t find a target id.'));
      }

      resolve(target.id);
    });
  });
});
```